app/components/ChangeLog.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMemo, useState } from "react";
+import Card from "react-bootstrap/Card";
+import Table from "react-bootstrap/Table";
+import { formatDateTime } from "@/utils/formatters";
+import ChangeLogDetails from "./ChangeLogDetails";
+import {
+  ChangeLogEntry,
+  ChangeLogDiff,
+  ChangeLogEntryEnriched,
+} from "@/types/types";
+
+const KEYS_TO_IGNORE = ["updated_at", "updated_by", "position"];
+
+/**
+ * Return an array of values that are different between the `old` object and `new` object
+ * Each object in the array takes the format
+ * { field: <property name>, old: <old value>, new: <new value> }
+ */
+const getDiffArray = <T extends Record<string, unknown>>(
```

Comment:
getting this typed correctly was correctly, even though the result looks simple. these kind of data-munging tasks are what i've found to be the most onerous to convert to typescript.

Reply:
also, this is the first of many declarations of `Record<string, unknown>>` because the `ColBaseDef` extends from it. 

i _think_ it's right to extend from this type, but because it is used so often we should probably give it a name and export it. something like `<RecordWithStringKeys>`. that's all it is: an object with keys that are always strings.

this will make more sense after reading about the `Record` utility type: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type  :)